src/domain/Tables/Table/subcomponents/TableRow.tsx

```diff
@@ -28,80 +24,58 @@ import {
   StyledSwipeActionsButtonWrapper,
   StyledSwipeActionsCell
 } from '../services/style'
-import { IColumnProps, IRowProps, ISelectedRows, Table, getActionsIconButtonGroup } from '../Table'
+import {
+  IColumnProps,
+  IRowProps,
+  InteractionType
+} from '../services/types'
+import { getActionsIconButtonGroup } from './ActionIconButtonGroup'
 import { TableCheckboxInput, TableCheckboxInputValue } from './TableCheckboxInput'
+import { TableRowDataCell } from './TableRowDataCell'
 
 interface ITableRowProps <T> {
   columns: Array<IColumnProps<T>>
   row: IRowProps<T>
-  selectedAll: TableCheckboxInputValue
-  selectedRows: ISelectedRows
-  setSelectedRows: (value: ISelectedRows) => void
-  interactionType: Table.InteractionType
+  isSelected: boolean
+  toggleSelected: (row: IRowProps<T>) => void
+  interactionType: InteractionType
```

Comment:
`selectedRows` meant that whenever any row was selected, every row in the table had to rerender. Changing this to `isSelected` means that only the clicked row needs to change.